src/profiler.cpp

```diff
@@ -1647,6 +1652,124 @@ void Profiler::dumpText(Writer& out, Arguments& args) {
     }
 }
 
+/*
+ * If the value has been seen already, return its idx in `values`.
+ * If not, update the map and return the index.
+ */
+static inline u32 getOrSetIndex(
+        std::unordered_map<std::string, u32>& idx_map,
+        const std::string& value
+) {
+    return idx_map.insert({value, idx_map.size()}).first->second;
+}
```

Comment:
Can we have a class (let's say, `Index`, similarly to `one.convert.Index` in Java) that encapsulates this logic?
This class could be probably reused for Element names in `jfrMetadata`.

And as I mentioned in a previous review, `getOrSet` name is not good. What the function does is unambiguous: it always obtains an index of a string, there is no "or" in its behavior. The fact that the function updates internal map as a side effect is an implementation detail.

Reply:
Can you check 0ea647b581b0b7be0f0155459fa9bf62c08b741a @apangin ?